### DISCUSSION TITLE
Window click not working

### DISCUSSION THREAD

**Discussion by zSynctic** [spyoungtech/ahk#254] [Q&A] [answered]
I'm not sure if its the click that is not working or its the coordinates because I saw another discussion and the answer was that it clicks based on the window not the screen coordinates if thats the issue how can I get the coordinates from the window. Thanks

Here is my code which is a test for the library:
```
from ahk import AHK

ahk = AHK()

ahk.set_coord_mode('Mouse', 'Screen')

win = ahk.find_window(title='New chat - Google Chrome')
assert win is not None, "Window not found"

# click within in the window
win.click(x=216, y=122)
```

**Comment by spyoungtech** [comment, marked as answer]
The `Window.click` method uses [ControlClick](https://www.autohotkey.com/docs/v2/lib/ControlClick.htm) under the hood. So, it obeys the relative positioning considerations mentioned in the docs there. That is:

> Mode 1 (Position): Specify the X and Y coordinates relative to the upper left corner of the target window's [client area](https://www.autohotkey.com/docs/v2/lib/CoordMode.htm#Client).

If the window you're interacting with is in the foreground, you may find using `ahk.click` easier if you want to use coordinates relative to the screen rather than the window.

By default, `.get_mouse_position()` (or `.mouse_position`) will retrieve coordinates relative to the _active_ window. This is also the default behavior for `ahk.click` (which uses [Click](https://www.autohotkey.com/docs/v2/lib/Click.htm)) but you can set the coordinate mode to change this behavior.

From the examples in the README:
```py
ahk.mouse_position  # Returns a tuple of mouse coordinates (x, y) (relative to active window)
ahk.get_mouse_position(coord_mode='Screen') # get coordinates relative to the screen
ahk.click(200, 200)  # Moves the mouse to a particular position and clicks (relative to active window)
ahk.click(100, 200, coord_mode='Screen') # click relative to the screen instead of active window
```


Does that help answer your question?
